src/controllers/things_controller.js

```diff
@@ -223,15 +223,34 @@ ThingsController.patch('/:thingId', function(request, response) {
       request.body.floorplanX, request.body.floorplanY);
   }).then((description) => {
     response.status(200).json(description);
-  }).catch(function(e) {
-    response.status(500).send(`Failed to update thing ${thingId} ${e}`);
+  }).catch((e) => {
+    response.status(500).send(`Failed to update thing ${thingId}: ${e}`);
+  });
+});
+
+/**
+ * Modify a Thing.
+ */
+ThingsController.put('/:thingId', (request, response) => {
+  const thingId = request.params.thingId;
+  if (!request.body || !request.body.hasOwnProperty('name')) {
+    response.status(400).send('name parameter required');
+    return;
+  }
+
+  Things.getThing(thingId).then((thing) => {
+    return thing.setName(request.body.name);
```

Comment:
While prohibited in the UI, I can manually make a request through curl which sets the thing's name to '' or ' ' which isn't that big a deal but would be nice to prohibit.